src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,402 @@
+import {Counters} from "openzeppelin-contracts/contracts/utils/Counters.sol";
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "openzeppelin-contracts/contracts/security/Pausable.sol";
+import {Wiki} from "../../src/Wiki.sol";
+
+interface IERC20 {
+    function transferFrom(
+        address from,
+        address to,
+        uint256 amount
+    ) external returns (bool);
+
+    function balanceOf(address account) external view returns (uint256);
+
+    function transfer(address to, uint256 amount) external returns (bool);
+}
+
+/// @title BRAIN Pass NFT
+/// @author Oleanji
+/// @notice A pass for IQ Wiki Editors
+
+contract BrainPassCollectibles is ERC721, Pausable, Ownable {
+    /// -----------------------------------------------------------------------
+    /// Errors
+    /// -----------------------------------------------------------------------
+
+    error MintingPaymentFailed();
+    error IncreseTimePaymentFailed();
+    error AlreadyMintedAPass();
+    error NotTheOwnerOfThisNft();
+    error InvalidMaxTokensForAPass();
+    error PassTypeNotFound();
+    error PassMaxSupplyReached();
+    error NoEtherLeftToWithdraw();
+    error TransferFailed();
+    error DurationNotInTimeFrame();
+    error PassTypeIsPaused();
+    error NoIQLeftToWithdraw();
+
+    /// -----------------------------------------------------------------------
+    ///  Inheritances
+    /// -----------------------------------------------------------------------
+    using Counters for Counters.Counter;
+
+    /// -----------------------------------------------------------------------
+    /// Structs
+    /// -----------------------------------------------------------------------
+
+    struct UserPassItem {
+        uint256 tokenId;
+        uint256 passId;
+        uint256 startTimestamp;
+        uint256 endTimestamp;
+    }
+
+    struct PassType {
+        uint256 passId;
+        string name;
+        uint256 pricePerDay;
+        uint256 maxTokens;
+        uint256 discount;
+        uint256 lastMintedId;
+        bool isPaused;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Mappings
+    /// -----------------------------------------------------------------------
+
+    mapping(uint256 => PassType) public passTypes;
+    mapping(address => UserPassItem) internal addressToNFTPass;
+
+    /// -----------------------------------------------------------------------
+    /// Constant
+    /// -----------------------------------------------------------------------
+
+    address public iqToken;
+
+    /// -----------------------------------------------------------------------
+    /// Variables
+    /// -----------------------------------------------------------------------
+
+    Counters.Counter private passIdTracker;
+    Counters.Counter private tokenIdTracker;
+    uint256 public MINT_LOWER_LIMIT = 28;
+    uint256 public MINT_UPPER_LIMIT = 365;
+    string public baseTokenURI;
```

Comment:
you probl. dont need this. 

erc721 from oz has a [_setBaseURI](https://docs.openzeppelin.com/contracts/3.x/api/token/erc721#ERC721-_setBaseURI-string-) and https://docs.openzeppelin.com/contracts/3.x/api/token/erc721#ERC721-baseURI--

Reply:
function _baseURI() internal view virtual override returns (string memory) {
        return baseTokenURI;
    }
    then ill have to set the url from here like
      function _baseURI() internal view virtual override returns (string memory) {
        return "https//food.com";
    }

Reply:
you are right. lets keep it 👍🏻